sound/soc/sof/sof-audio.c

```diff
@@ -189,12 +190,25 @@ int sof_restore_pipelines(struct device *dev)
 		switch (swidget->id) {
 		case snd_soc_dapm_dai_in:
 		case snd_soc_dapm_dai_out:
+			ipc_size = sizeof(struct sof_ipc_comp_dai) +
+				   sizeof(struct sof_ipc_comp_new_ext);
+			comp = kzalloc(ipc_size, GFP_KERNEL);
+			if (!comp)
+				return -ENOMEM;
+
```

Comment:
@keyonjie at the end of sof_widget_load_dai() the contents of dai->comp_dai already includes the ext data too, right? This step seems unnecessary while resuming. Simply using dai->comp_dai should suffice

Reply:
@ranj063 dai->comp_dai contains only the sizeof(struct sof_ipc_comp_dai) so the ext data is not stored there, that's why we need to retrieve the ext data from the swidget.

Reply:
@keyonjie are you saying this step here doesnt copy the ext_data?
https://github.com/thesofproject/linux/blob/6fc20386a9a98df355a2d1569773fc614e7dc536/sound/soc/sof/topology.c#L1505

But it is allocated the whole size here?
https://github.com/thesofproject/linux/blob/6fc20386a9a98df355a2d1569773fc614e7dc536/sound/soc/sof/topology.c#L1460

Reply:
@ranj063 exactly, it copy the comp_dai only, and the ext-data is copied subsequently

Reply:
@keyonjie then why not change it to  memcpy(&dai->comp_dai, comp_dai, sizeof(ipc_size));

Reply:
@ranj063 the comp_dai doesn't contain the extended data, we need to copy the from the sof widget.

Reply:
@keyonjie we're talking past each other on this one. I'm asking why cant we save the ext data into comp_dai when we are parsing the topology?

Reply:
@ranj063 let me add some comments as discussed to explain it in the next version.